doc/userguide/rules/smb-keywords.rst

```diff
@@ -58,3 +58,48 @@ Examples::
 ``smb.ntlmssp_domain`` is a 'sticky buffer'.
 
 ``smb.ntlmssp_domain`` can be used as ``fast_pattern``.
+
+
+smb.version
+------------
+
+Keyword to match on SMB version.
+
+Examples::
+
+  alert smb $HOME_NET any -> any any (msg:"SMBv1 version rule"; smb.version:1; sid:1;)
+  alert smb $HOME_NET any -> any any (msg:"SMBv2 version rule"; smb.version:2; sid:2;)
```

Comment:
Something that can be useful knowledge for other documentation bits - or for this one, in case there are other changes to be incorporated:
if you use ``.. container:: example-rule``, you can then emphasize specific parts of the rule:
![image](https://github.com/OISF/suricata/assets/16976857/0f526520-5c59-418e-88c4-8e78eab99a9a)
As seen here: https://docs.suricata.io/en/latest/rules/header-keywords.html#ttl

Non-formatted text looks like:
```
Example of the ttl keyword in a rule:

.. container:: example-rule

    alert ip $EXTERNAL_NET any -> $HOME_NET any (msg:"IP Packet With TTL 0";
    :example-rule-emphasis:`ttl:0;` classtype:misc-activity; sid:1; rev:1;)
    ```
    
    Apologies if this was already known!

Reply:
Ah thank you, I am not sure if I knew that or not. If I recall in this case I was trying to keep the same style within the file with existing examples. I am curious though if you think as documentation is updated if the new/better formatting would be worth including as part of any updates? Probably as its own commit though? If you think it merits another PR I would be happy to do so.

Reply:
To be very honest, I'm not sure what's the guideline we are following, for this. I think this is something that many either don't know about, or forget - myself included -, and therefore we have parts where we have this, and parts where it's not present. I think I'd like to return the question to you, as you're someone more likely to represent our audience for rule documentation:
How do you prefer to see a rule example, like it is for ttl, above, or like this, for example:
![image](https://github.com/OISF/suricata/assets/16976857/29dc1701-6b07-46c1-a73e-b854bb6354c0)

Reply:
Talking with others, the preference is the rule container, so if possible I think that would be the route to go.